data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/OpenSearchSink.java

```diff
@@ -141,6 +144,7 @@ public OpenSearchSink(final PluginSetting pluginSetting,
     this.awsCredentialsSupplier = awsCredentialsSupplier;
     this.sinkContext = sinkContext != null ? sinkContext : new SinkContext(null, Collections.emptyList(), Collections.emptyList(), Collections.emptyList());
     this.expressionEvaluator = expressionEvaluator;
+    this.latencyMetrics = new LatencyMetrics(pluginMetrics);
```

Comment:
Is there really no way to give the `EventHandle` class a copy of `PluginMetrics`, and then to just have the `releaseEventHandle` call automatically populate the metric so we don't have to pass them to every sink, and we keep to just one thing that everyone has to call (`releaseEventHandle`). You could even add to the function, and make it `releaseEventHandle(final EventHandle eventHandle, final String pluginId)`. The pluginId could just be the plugin name for now, but it can be used in the metric.

Reply:
How do we do that? Metrics are sink specific, right?

Reply:
Yes but we could "fake" that the metrics are owned by the sink by adding the name of the sink here (since you can name the metrics anything and that it can still follow our metric naming pattern

Reply:
I do not think that's the clean way because an event handle can only have one name for metrics. How can it have multiple names? I think this is way cleaner.

Reply:
Every sink will need to provide this latency and we want to make it as easy as possible for sinks to use this feature.

I think that data-prepper-core could hold a `LatencyMetrics` per `Sink`. Just before calling `Sink::output`, data-prepper-core registers an `onRelease` method similar to what I suggested in another comment. Then it receives the callback when `release` is called.

One thing we would need to be sure to do here is ensure that sinks always call `release`. This is possible now that  #3546 is merged.